test/helpers/framework.go

```diff
@@ -260,19 +260,25 @@ func (tc *E2ETestFramework) waitForElasticsearchPods(retryInterval, timeout time
 			clolog.Error(err, "Error listing elasticsearch pods")
 			return false, nil
 		}
-		if len(pods.Items) == 0 {
+		numberOfPods := len(pods.Items)
+		if numberOfPods == 0 {
 			clolog.V(2).Info("No elasticsearch pods found ", "pods", pods)
 			return false, nil
 		}
-
+		containersReadyCount := 0
 		for _, pod := range pods.Items {
 			for _, status := range pod.Status.ContainerStatuses {
 				clolog.V(3).Info("Checking status of", "PodName", pod.Name, "ContainerID", status.ContainerID, "status", status.Ready)
-				if !status.Ready {
-					return false, nil
+				if status.Ready {
+					containersReadyCount ++
 				}
 			}
 		}
+		if containersReadyCount >= numberOfPods {
```

Comment:
These numbers are not related other then there are N containers for every pod defined by ES.  I think we should rethink this check and we do a similar check in the functional framework that may be reusable here or able to be moved into a utility function

Reply:
@jcantrill @vimalk78 

Please advice what to compare as this issue is one of the big killers for e2e tests. If you think we should look for ES CR and read from that ... this is doable but let me know if you want to go that path ?

Reply:
Updated logic to make sure we have at least one container ready and no containers waiting ... this makes sure that the CR was read by the controller and it started to create containers and that there are no waiting containers ...

@vimalk78 @jcantrill please approve